src/rp2_common/hardware_sync/include/hardware/sync.h

```diff
@@ -54,22 +70,32 @@ typedef volatile uint32_t spin_lock_t;
 #define PICO_SPINLOCK_ID_HARDWARE_CLAIM 11
 #endif
 
-// PICO_CONFIG: PICO_SPINLOCK_ID_STRIPED_FIRST, Spinlock ID for striped first, min=16, max=31, default=16, group=hardware_sync
+// PICO_CONFIG: PICO_SPINLOCK_ID_OS1, Spinlock ID reserved for use by low level OS style software, min=0, max=31, default=12, group=hardware_sync
+#ifndef PICO_SPINLOCK_ID_OS1
+#define PICO_SPINLOCK_ID_OS1 14
+#endif
+
+// PICO_CONFIG: PICO_SPINLOCK_ID_OS2, Spinlock ID reserved for use by low level OS style software, min=0, max=31, default=13, group=hardware_sync
+#ifndef PICO_SPINLOCK_ID_OS2
+#define PICO_SPINLOCK_ID_OS2 15
+#endif
+
+// PICO_CONFIG: PICO_SPINLOCK_ID_STRIPED_FIRST, Lowest Spinlock ID in the 'striped' range, min=0, max=31, default=16, group=hardware_sync
```

Comment:
If it's now `min=0` (whereas previously it was `min=16`) does there need to be a preprocessor-check (or runtime assertion) that the striped spinlock range doesn't overlap with any of the fixed-purpose spinlocks? :shrug:  
Or is this just something that the developer needs to be careful of themselves?

Reply:
yes... frankly these are "advanced"

Reply:
Yep if you are reallocating spinlocks then you had best read the surrounding comments!